File: The Cosmic Era/P3 - Battle Positions.py
# https://dmoj.ca/problem/seed3
# The Cosmic Era P3 - Battle Positions

import sys
input = sys.stdin.readline
i = int(input())
n = int(input())
j = int(input())

arr = [0 for b in range(i+1)]

for itr in range(j):
    xi, xf, k = map(int, input().split())
    arr[xi-1] +=k
    arr[xf] -=k


for itr in range(1,i+1):
    arr[itr] += arr[itr-1]


ttl = 0

for a in range(i):
    if arr[a] < n:
        ttl+=1
print(ttl)

File: Dmoj/The Cosmic Era/P3 - Battle Positions.cpp
// https://dmoj.ca/problem/seed3
// The Cosmic Era P3 - Battle Positions

/********   All Required Header Files ********/
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <sstream>
#include <queue>
#include <deque>
#include <bitset>
#include <iterator>
#include <list>
#include <stack>
#include <map>
#include <set>
#include <functional>
#include <numeric>
#include <utility>
#include <limits>
#include <time.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>

using namespace std;

int i; int n; int j;

int psa[100005];

int main() {
    cin>> i >> n >> j;
    int xi; int xf; int k;
    for (int itr =0; itr<j; itr++) {
        cin >> xi >> xf >> k;
        psa[xi-1] +=k;
        psa[xf] -=k;
    }

    for (int itr = 0; itr < i; itr++){
        psa[itr] += psa[itr-1];
    }


    int ttl = 0;

    for (int a = 0; a<i; a++) {
        if (psa[a] < n){
            ttl+=1;
        }
    }
    cout << ttl;
    return(0);
}